src/stlink-lib/usb.c

```diff
@@ -1109,7 +1109,7 @@ int _stlink_usb_read_trace(stlink_t* sl, uint8_t* buf, size_t size) {
         return -1;
     }
 
-    size_t trace_count = read_uint16(sl->q_buf, 0);
+    uint32_t trace_count = read_uint16(sl->q_buf, 0);
```

Comment:
Why limit the bit width of a variable? Can unsigned be used?

Reply:
It is more easily support stable compilation for different bitness and many compilers & OSes when used values with a defined size.

Reply:
My opinion is that shouldn't use bounded types unnecessarily. It is easier for the system to work with native tipe (int).
ps You can ignore my remark. Everything will work well without any changes.

Reply:
> It is easier for the system to work with native tipe (int)

It is not true anymore. An unsigned has a 4-byte size for 32 and 64-bit platforms. But should be printed with different format flags.
For "real" platform fast you should use something like _int_fast32_t_.

Reply:
What are the differences in the flags? I always thought that "%u" it is unsigned.
int_fast32_t on most systems (32-bit and higher) is int. Int should be the most effective.

Reply:
My suggestion should still use the size_t for unsigned sizes or loop counters. But in this case we do a read_uint16 and should put it in a uint16.

Reply:
In my opinion we should use bounded types whenever possible _and_ useful.
Especially on transition from unbounded types to the latter there should be intensive testing on several operating system environments before submitting any PR to rule out regression bugs.

With the next release (simultaneously) I'll further introduce a proper testing/nightly branch as well.
I'm not so happy with the current situation of patching around on `develop` to get it working again (some people want to use it), while CI is buggy/not functional. However this is not meant to blame anyone - it solely describes the current state. :unamused: